docs/rfcs/0002-layer-architecture.md

```diff
@@ -0,0 +1,328 @@
+
+- Describe the details and/or technical implementations of the various layers. See the detailed RFCs for the layers if looking for that information.
+- Cover every minor component or implementation-specific components, this RFC is designed to only give an overview.
+
+## Proposal
+
+_This is the technical portion of the RFC. Explain the design in sufficient detail that:_
+
+- _Its interaction with other features is clear._
+- _It is reasonably clear how the feature would be implemented._
+- _Corner cases are dissected by example._
+
+_The section should return to the examples given in the guide-level explanation below, and explain more fully how the detailed proposal makes those examples work._
+
+Racklet is divided into 5 distinct layers, from lowest-level to highest-level:
+
+1. **Structural**
+2. **Electrical**
+3. **Firmware**
+4. **System Software**
+5. **User Software**
+
+There is some overlap between these defined layers, mostly due to individual components contributing to multiple layers, but we aim to keep a clear distinction in this definition. If for example a microcontroller is part of both the electrical and firmware layer, the electrical layer only considers its electical properties and the firmware layer only its firmware.
+
+The architecture is designed with the layers and their interaction as the primary focus. The requirements of a layer drive the design of the layer below it, which aims to satisfy the dependencies according to the values and user goals of the project. The layers are described here in reverse order (layer 5 first), since the highest layer starts the dependency chain by directly fulfilling the user goals.
+
+### 5. User Software layer
+
+**Summary**: The user software layer should allow the user to schedule workloads of choice using either containers or VMs. There should be an accessible and observable graphical user interface in place for the user to monitor and manage the Racklet system and workloads.
+
+**Goals**:
+- Enable the user to observe and manage a Racklet cluster
+- Enable easy deployment of container/VM workloads
+- (Optionally) make a Kubernetes cluster accessible for the user
+
+**Layer components**:
+
+| Component                     | Role                                             | Key Requirements                                                                                        |
+| -------------------------------- | ------------------------------------------------ | ------------------------------------------------------------------------------------------------------- |
+| **VM image building automation** | Define and run VMs declaratively                 | [Improve status quo], [Openness], [Declarative management], [Documentation], [Fast reconfiguration]     |
+| **Kubernetes VM image**          | Consume/use a Kubernetes cluster                 | [De-facto standards], [Declarative management], [Loose coupling], [Upgradability], [Utilize Kubernetes] |
```

Comment:
I'd omit "VM image" here as an implementation detail, and describe it instead as a system of components, which it is gonna be at the end of the day, so e.g. `Kubernetes deployment automation`